### ISSUE TITLE
Incorrect warning：#[warn(unused_variables)]

### ISSUE THREAD

**Issue by ubeing** [rust-lang/rust-analyzer#18534]
**rust-analyzer version**: V0.3.2188

**rustc version**: 1.81.9

**editor or extension**: VSCode on macOS

**code snippet to reproduce**: 
```rust
pub async fn new_course(
    new_course: web::Json<Course>,
    app_state: web::Data<AppState>,
) -> HttpResponse {
        let course_count = app_state
        .courses
        .lock()
        .unwrap()
        .clone()
        .into_iter()
        .filter(|course| course.teacher_id == new_course.teacher_id)
        .collect::<Vec<Course>>()
        .len();
    let new_course = Course {
        teacher_id: new_course.teacher_id,
        id: Some(course_count + 1),
        name: new_course.name.clone(),
        time: Some(Utc::now().naive_utc()),
    };
    app_state.courses.lock().unwrap().push(new_course);
    HttpResponse::Ok().json("Course added")
}

```
The IDE always issues an unused variable warning for the variable 'course_count', even if you prefix the variable with '_'

**Comment by lnicola** [comment]
Which crates are you using there?

**Comment by lnicola** [comment]
FWIW, I don't get any warning:

![Image](https://github.com/user-attachments/assets/aa7bef02-0557-4639-9e55-c7183007b80e)
